commit d64b91a6d4fa1a3ac1d01fa7593d67c5e5ed5101
Author: Yann E. MORIN <yann.morin.1998@anciens.enib.fr>
Date:   Sun Jan 9 01:45:04 2011 +0100

    ARM: reorder "Use BX" option
    
    "Use BX" is not available on all CPUs, so the option depends on
    a correct CPU to be chosen . It is weird that e BX" then appears
    _above_ the CPU selection, not below.
    
    Move the "Use BX" after the CPU selection.
    
    Signed-off-by: "Yann E. MORIN" <yann.morin.1998@anciens.enib.fr>
    Cc: Khem Raj <raj.khem@gmail.com>
    Cc: Bernhard Reutner-Fischer <rep.dot.nop@gmail.com>
    Cc: Carmelo AMOROSO <carmelo.amoroso@st.com>
    Signed-off-by: Khem Raj <raj.khem@gmail.com>
    Signed-off-by: Bernhard Reutner-Fischer <rep.dot.nop@gmail.com>

diff --git a/extra/Configs/Config.arm b/extra/Configs/Config.arm
index eb27a33..dd63b80 100644
--- a/extra/Configs/Config.arm
+++ b/extra/Configs/Config.arm
@@ -30,13 +30,6 @@ config CONFIG_ARM_EABI
 
 endchoice
 
-config USE_BX
-	bool "Use BX in function return"
-	default y
-	depends on !CONFIG_GENERIC_ARM && !CONFIG_ARM610 && !CONFIG_ARM710
-	help
-	  Use BX instruction for THUMB aware architectures.
-
 choice
 	prompt "Target Processor Type"
 	default CONFIG_GENERIC_ARM
@@ -131,3 +124,10 @@ config CONFIG_ARM_IWMMXT
 	select ARCH_HAS_MMU
 
 endchoice
+
+config USE_BX
+	bool "Use BX in function return"
+	default y
+	depends on !CONFIG_GENERIC_ARM && !CONFIG_ARM610 && !CONFIG_ARM710
+	help
+	  Use BX instruction for THUMB aware architectures.
